Content/GUI/Utilities/UISimpleWrappableText.cs

```diff
@@ -0,0 +1,322 @@
+	{
+		foreach (Vector2 t in ChatManager.ShadowDirections)
+		{
+			DrawColorCodedString(spriteBatch, font, snippets, position + t * spread,
+				baseColor, rotation, origin, baseScale, out int _, maxWidth, true);
+		}
+	}
+
+	private static Vector2 DrawColorCodedString(SpriteBatch spriteBatch, DynamicSpriteFont font, TextSnippet[] snippets,
+		Vector2 position, Color baseColor, float rotation, Vector2 origin, Vector2 baseScale, out int hoveredSnippet,
+		float maxWidth, bool ignoreColors = false)
+	{
+		int num = -1;
+		var vector21 = new Vector2(Main.mouseX, Main.mouseY);
+		Vector2 x = position;
+		Vector2 vector22 = x;
+		float single = font.MeasureString(" ").X;
+		Color visibleColor = baseColor;
+		float single1 = 0f;
+		for (int i = 0; i < snippets.Length; i++)
+		{
+			TextSnippet textSnippet = snippets[i];
+			textSnippet.Update();
+			if (!ignoreColors)
+			{
+				visibleColor = textSnippet.GetVisibleColor();
+			}
+
+			float scale = textSnippet.Scale;
+			if (!textSnippet.UniqueDraw(false, out Vector2 vector2, spriteBatch, x, visibleColor, scale))
+			{
+				string[] strArrays = textSnippet.Text.Split(['\n']);
+				string[] strArrays1 = strArrays;
+				for (int j = 0; j < strArrays1.Length; j++)
+				{
+					string[] strArrays2 = strArrays1[j].Split([' ']);
+					for (int k = 0; k < strArrays2.Length; k++)
+					{
+						if (k != 0)
+						{
+							ref float singlePointer = ref x.X;
+							singlePointer += single * baseScale.X * scale;
+						}
+
+						if (maxWidth > 0f &&
+						    x.X - position.X + font.MeasureString(strArrays2[k]).X * baseScale.X * scale > maxWidth)
+						{
+							x.X = position.X;
+							ref float y = ref x.Y;
+							y += font.LineSpacing * single1 * baseScale.Y;
+							vector22.Y = Math.Max(vector22.Y, x.Y);
+							single1 = 0f;
+						}
+
+						if (single1 < scale)
+						{
+							single1 = scale;
+						}
+
+						DynamicSpriteFontExtensionMethods.DrawString(spriteBatch, font, strArrays2[k], x, visibleColor,
+							rotation, origin, baseScale * textSnippet.Scale * scale, SpriteEffects.None, 0f);
+						Vector2 vector23 = font.MeasureString(strArrays2[k]);
+						if (vector21.Between(x, x + vector23))
+						{
+							num = i;
+						}
+
+						ref float x1 = ref x.X;
+						x1 += vector23.X * baseScale.X * scale;
+						vector22.X = Math.Max(vector22.X, x.X);
+					}
+
+					if (strArrays.Length <= 1 || j >= strArrays1.Length - 1)
+					{
+						continue;
+					}
+
+					ref float lineSpacing = ref x.Y;
+					lineSpacing += font.LineSpacing * single1 * baseScale.Y;
+					x.X = position.X;
+					vector22.Y = Math.Max(vector22.Y, x.Y);
+					single1 = 0f;
+				}
+			}
+			else
+			{
+				if (vector21.Between(x, x + vector2))
+				{
+					num = i;
+				}
+
+				ref float singlePointer1 = ref x.X;
+				singlePointer1 += vector2.X * baseScale.X * scale;
+				vector22.X = Math.Max(vector22.X, x.X);
+			}
+		}
+
+		hoveredSnippet = num;
+		return vector22;
+	}
+}
```

Comment:
Moreso a question than a comment - vanilla's `UIText` is already wrappable, using the `IsWrapped` property. What exactly does this do that `UIText` can't? I wouldn't be able to tell by just reading this code.

Reply:
The larger part of this was copied from Spirit mod as it was used by the UISelectableQuest - Being unfamiliar with UIText I'm unsure what this does over it.

Reply:
I'll try and see if they're interchangeable real quick, if not then I'll keep it as is.

Reply:
Going to leave an additional comment as having no quest will invariably crash the game due to how QuestsUIState is programmed.

Reply:
![image](https://github.com/Path-of-Terraria/PathOfTerraria/assets/20131629/9115b1b9-f4ef-488d-935c-2b8f04704ffd)
Won't be testing the UI yet, the UI is always like this and it makes it really annoying to test, especially since it doesn't block mouse input.

Reply:
Yeah I had it in the center previously but the changes catNull made put it in the top left and I never adjusted it back. Wasn't sure if that was intended or not. 

Not sure how to block UI mouse input but can look tomorrow.

Reply:
`Main.blockInput` *might* be it, but I've never fully understood blocking mouse input. It's surprisingly unclear lol

Reply:
I ended up adding a whole dynamic positioning setup like I did with the other UI stuff. I plan try and move to a percent based instead of specific positions later but this will do for now

Reply:
Alright, so with some testing I'm mostly confused. This should block mouse input just fine:
```cs
		Main.blockMouse = true;
		Main.isMouseLeftConsumedByUI = true;
		Main.LocalPlayer.mouseInterface = true;
```
But the Quest UI's dimensions are super off, the visual is seemingly unrelated to where and how large the actual UI state is. This means blocking input is functionally useless. This _will_ cause issues later since UI is very dependent on the given Width, Height, Left, Top, HAlign and VAlign fields and all of these are currently being completely ignored. The whole UI seems really awkwardly positioned as a result. I don't know where this issue stems from, as I don't know SmartUIState enough to read the UI tree, but some notable issue is happening here.
As for the `UISelectableQuest` thing, I don't think it's worth spending dev time on. Especially not with the above issues.

Reply:
I see. 
![image](https://github.com/Path-of-Terraria/PathOfTerraria/assets/24656209/769c74c2-ab92-4409-a9d6-d28f54c8c927)
I wasn't thinking about that being effected. I am working on some debugging tools and also rewriting of this to accommodate this.